tests/components/sensor/test_entur_public_transport.py

```diff
@@ -0,0 +1,67 @@
+"""The tests for the tube_state platform."""
+from datetime import datetime
+import unittest
+from unittest.mock import patch
+
+from enturclient.api import RESOURCE
+from enturclient.consts import ATTR_EXPECTED_AT, ATTR_ROUTE, ATTR_STOP_ID
+import requests_mock
+
+from homeassistant.components.sensor.entur_public_transport import (
+    CONF_EXPAND_PLATFORMS, CONF_STOP_IDS)
+from homeassistant.setup import setup_component
+import homeassistant.util.dt as dt_util
+
+from tests.common import get_test_home_assistant, load_fixture
+
+VALID_CONFIG = {
+    'platform': 'entur_public_transport',
+    CONF_EXPAND_PLATFORMS: False,
+    CONF_STOP_IDS: [
+        'NSR:StopPlace:548',
+        'NSR:Quay:48550',
+    ]
+}
+
+FIXTURE_FILE = 'entur_public_transport.json'
+TEST_TIMESTAMP = datetime(2018, 10, 10, 7, tzinfo=dt_util.UTC)
+
+
+class TestEnturPublicTransportSensor(unittest.TestCase):
+    """Test the tube_state platform."""
+
+    def setUp(self):
+        """Initialize values for this testcase class."""
+        self.hass = get_test_home_assistant()
+        self.config = VALID_CONFIG
+
+    def tearDown(self):
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    @requests_mock.Mocker()
+    @patch(
+        'homeassistant.components.sensor.entur_public_transport.dt_util.now',
+        return_value=TEST_TIMESTAMP)
+    def test_setup(self, mock_req, mock_patch):
```

Comment:
The argument order should be from inner to outer patch decorator.

Reply:
If I'm understanding you correctly, I should reorder the arguments to have an order that matches patches pattern where the lowest decorator is first of the arguments. However when I tested it out it gives the mock as input on the wrong argument. Seems like the ordering doesn't follow that pattern when it's a requests mocker on top. From what I could see, when I tried to figure out how others have done it within HA, the ordering it has now is the common way to do it. 
Is it something I'm missing here?

Reply:
It seems the requests_mock mocker has a special behavior and its argument is always placed first (contrary to its [docs](https://requests-mock.readthedocs.io/en/latest/mocker.html) :man_shrugging:). Normally patch decorators are passed in the order I stated above.

https://docs.python.org/3/library/unittest.mock.html#nesting-patch-decorators